### DISCUSSION TITLE
Match not found with consult-ripgrep and consult-grep

### DISCUSSION THREAD

**Discussion by bmp** [minad/consult#1031] [Q&A]
Hello,

Thank you for the wonderful package and apologies if this is not the correct place for this question (I did try searching for this over the web with no success).

`M-x grep` & `M-x ripgrep-regexp` both return accurate search results for me. However, when I try either `consult-grep` or `consult-ripgrep` I only get a message saying 'match required' and the command is not executed. 

I am using the default configuration for consult found in the [Readme](https://github.com/minad/consult?tab=readme-ov-file#use-package-example).

Thank you!

_Edited for a typo_

**Comment by minad** [comment]
Please see https://github.com/minad/consult?tab=readme-ov-file#bug-reports. Which completion UI do you use?

**Reply by bmp** [reply]
I use `vertico-mode` and `orderless`, when I run with the following configuration as recommended in the readme I can see ripgrep working.

```
(package-initialize)
(require 'consult)
(require 'vertico)
(vertico-mode)
(setq completion-styles '(substring basic))
```
Is there anything specific I need to check? In addition, when I deleted `consult` and re-installed it with my configuration I also noticed the following message in the message buffer,
`rg: ./source/emacs-29.3/test/manual/etags/a-src/empty.zz.gz:`.

**Reply by minad** [reply]
You can check the ` *consult-async*` log buffer for errors. See the README.

**Reply by bmp** [reply]
Apologies, the buffer is not created at all as I think the command isn't executed. I have attached a screenshot showing what I mean below.

![Screenshot_20240531_214054](https://github.com/minad/consult/assets/1095177/c19da3e7-8391-4523-8ac9-a180db5bdf44)

This is the only message I can see in the `*Messages*` buffer,
```
/usr/bin/grep: ./.local/share/containers/storage/overlay/778589c7af044c643232251405029863789e264a2cb22157792a341e3058ce0b: Permission denied
```

**Reply by minad** [reply]
This looks like an installation issue, which is not related to Consult.

**Reply by bmp** [reply]
Oh! Could you point me to any other resources? I am on OpenSuse Tumbleweed and I compiled emacs 29.3 myself with pgtk.

**Reply by bmp** [reply]
I found another [StackExchange](https://emacs.stackexchange.com/questions/74462/how-to-properly-use-consult-ripgrep-to-search-through-org-roam-notes) post with the same issue and based on the comments there I tried the following,
1. The options provided there to redifine the consult-ripgrep-command and run it, I get the same error.
2. Ran `rg --multiline --null --ignore-case --type org --line-buffered --color=always --max-columns=500 --no-heading --line-number . -e tasting ~/` in the command line and it does return valid results. but the same command when run through consult-ripgrep doesn't work.
3. I have also executed the command in Emacs with [dwim-shell-command](https://github.com/xenodium/dwim-shell-command) and I do get the results as expected.
4. Additionally, the search works with [deadgrep](https://github.com/Wilfred/deadgrep) which internally uses `rg` as well.

I am at a loss to figure out the problem which persists with `consult-ripgrep`  and `consult-grep` now!

**Reply by OrionRandD** [reply]
> I found another [StackExchange](https://emacs.stackexchange.com/questions/74462/how-to-properly-use-consult-ripgrep-to-search-through-org-roam-notes) post with the same issue and based on the comments there I tried the following,
> 
> 1. The options provided there to redifine the consult-ripgrep-command and run it, I get the same error.
> 2. Ran `rg --multiline --null --ignore-case --type org --line-buffered --color=always --max-columns=500 --no-heading --line-number . -e tasting ~/` in the command line and it does return valid results. but the same command when run through consult-ripgrep doesn't work.
> 3. I have also executed the command in Emacs with [dwim-shell-command](https://github.com/xenodium/dwim-shell-command) and I do get the results as expected.
> 4. Additionally, the search works with [deadgrep](https://github.com/Wilfred/deadgrep) which internally uses `rg` as well.
> 
> I am at a loss to figure out the problem which persists with `consult-ripgrep` and `consult-grep` now!

I also had this problems with consult-fd, consult-grep, consult-ripgrep, et.al - and after disabling ivy-mode and ido-mode, everything works great now. See here:

;; https://github.com/minad/consult?tab=readme-ov-file#bug-reports

;; Either use the default completion UI or ensure that exactly one of
;; vertico-mode, mct-mode, or icomplete-mode is enabled.

;; The unsupported modes that *MUST BE DISABLED*:

;; *helm-mode*,
;; *ido-mode* 
;; *ido-ubiquitous-mode*
;; *ivy-mode*,
;; *selectrum-mode*,

**Reply by OrionRandD** [reply]
I just would like to write a function for disabling this modes automatically when calling consult and returning to use ido-mode and ivy-mode after calling consult... instead of enabling it and disabling it by-hand...
